src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogPage.cs

```diff
@@ -758,26 +761,28 @@ internal void Validate()
                 if (standardButton.IsCreated)
                 {
                     buttonFlags |= standardButton.GetStandardButtonFlag();
-
-                    if (standardButton.DefaultButton && defaultButtonID == 0)
-                    {
-                        defaultButtonID = standardButton.ButtonID;
-                    }
                 }
             }
 
             for (int i = 0; i < _boundCustomButtons.Length; i++)
             {
                 TaskDialogButton customButton = _boundCustomButtons[i];
                 flags |= customButton.Bind(this, CustomButtonStartID + i);
+            }
 
-                if (customButton.IsCreated)
+            if (DefaultButton != null)
+            {
+                // ensure the default button is part of the button collection
+                if (!buttons.Contains(DefaultButton))
                 {
-                    if (customButton.DefaultButton && defaultButtonID == 0)
-                    {
-                        defaultButtonID = customButton.ButtonID;
-                    }
+                    throw new InvalidOperationException(SR.TaskDialogDefaultButtonMustExistInCollection);
                 }
+
+                defaultButtonID = DefaultButton.ButtonID;
```

Comment:
I think if the button is not created (e.g. if `Visible == false`), we must not use its `ButtonID` as it will not be part of the native task dialog.
```suggestion
                if (DefaultButton.IsCreated)
                {
                    defaultButtonID = DefaultButton.ButtonID;
                }
```